### PR TITLE
39 integrate minigame frame and car switches

### PR DESCRIPTION
**Pull request by LeonUwe**
#39 

When a car switch / hack is attempted, both players are sent to a random minigame. The winner of the minigame gets (to keep) the physical car.

A short message is displayed after the minigame is done/cancelled in the minigame ui. This makes it clear to the players who won and what will happen next.

### CODE REVIEW

**Comment by githubgedoens** [comment]
When I try to hack another player, I get an error: target_vehicle.get_player_id() is None. Is it possible that vehicle_in_proximity is currently never initialized?

**Comment by LeonUwe** [comment]
> When I try to hack another player, I get an error: target_vehicle.get_player_id() is None. Is it possible that vehicle_in_proximity is currently never initialized?

Where exactly did you get that error? Please post the whole error message. The player id of the target vehicle should not really matter...

**Comment by githubgedoens** [comment]
> > When I try to hack another player, I get an error: target_vehicle.get_player_id() is None. Is it possible that vehicle_in_proximity is currently never initialized?
> 
> Where exactly did you get that error? Please post the whole error message. The player id of the target vehicle should not really matter...

IAV_Distortion\src\UserInterface\DriverUI.py", line 212, in switch_cars
    target_player = target_vehicle.get_player_id()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get_player_id'

I tried this with both cars having drivers.

**Comment by LeonUwe** [comment]
> > > When I try to hack another player, I get an error: target_vehicle.get_player_id() is None. Is it possible that vehicle_in_proximity is currently never initialized?
> > 
> > 
> > Where exactly did you get that error? Please post the whole error message. The player id of the target vehicle should not really matter...
> 
> IAV_Distortion\src\UserInterface\DriverUI.py", line 212, in switch_cars target_player = target_vehicle.get_player_id() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ AttributeError: 'NoneType' object has no attribute 'get_player_id'
> 
> I tried this with both cars having drivers.

I can't replicate the error. I added some checks for incomplete information in the switch car event.

File: src/UserInterface/templates/minigame_index.html
Comment (githubgedoens) [inline src/UserInterface/templates/minigame_index.html:1]: If the virtual vehicle loses, the other player does not overtake the virtual verhicle

**Review by githubgedoens — APPROVED** [review]
Sometimes the physical car has a functionless hack button -> should be fixed in a future bug fix

File: src/UserInterface/templates/minigame_index.html
Comment (LeonUwe) [reply src/UserInterface/templates/minigame_index.html:1]: Yes, this is correct behaviour and is now ([d0bd6ac](https://github.com/LeonUwe/IAV_Distortion/pull/40/commits/d0bd6aca966227c4b8d1fc8aeebc51633da0f1a1)) correctly reflected in the messages the players receive after a minigame

**Comment by LeonUwe** [comment]
> Sometimes the physical car has a functionless hack button -> should be fixed in a future bug fix

Bug Issue has been created. Please add any more information you have on how to replicate this bug here #42.

**Pull request merged**